lib/velum/instance_type.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+require "json"
+
+module Velum
+  # Describe a public cloud instance type
+  class InstanceType
+    attr_reader :sort_value, :key, :category, :vcpu_count, :ram_bytes,
+      :ram_si_units, :details
+
+    def initialize(key, raw_data)
+      @key = key
+      @sort_value = raw_data["sort_value"]
+      @category = Velum::InstanceCategory.new(raw_data["category"])
+      @vcpu_count = raw_data["vcpu_count"]
+      @ram_bytes = raw_data["ram_bytes"]
+      @ram_si_units = raw_data["ram_si_units"]
+      @details = raw_data["details"]
+    end
+
+    def <=>(other)
+      sort_value <=> other.sort_value
+    end
+
+    def self.for(cloud)
+      raw_collection = get_raw_collection(cloud)
+      instance_type_collection_factory(
+        raw_collection["instance_types"],
+        raw_collection["categories"]
+      )
+    end
+
+    private
```

Comment:
i think in you need to write it like this, otherwise the `private` has no effect:

```ruby
    class << self
      def for(cloud)
        raw_collection = get_raw_collection(cloud)
        instance_type_collection_factory(
          raw_collection["instance_types"],
          raw_collection["categories"]
        )
      end

      private

      def get_raw_collection(cloud)
        data_path = Rails.root.join("config", "data", "#{cloud}-types.json")
        JSON.parse(File.read(data_path))
      end

      def instance_type_collection_factory(raw_types, raw_categories)
        raw_types.collect do |key, values|
          values["category"] = raw_categories[values["category"]]
          InstanceType.new(key, values)
        end.sort!
      end
    end
```

Reply:
Yeah, I saw the rubocop complaint; already pushed a change.